File: 1-TwoSum/1-TwoSum.py
# Last updated: 4/22/2025, 6:02:14 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        o = 0
        for outer in nums:
            i = o + 1
            for inner in nums[i:]:
                if outer + inner == target:
                    return [o, i] 
                i += 1
            o += 1

File: 1-TwoSum/1-TwoSum.go
// Last updated: 24/07/2025, 12:49:53
// @param nums: array of integers, can be negative
// @param target: sum target, can be negative
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int)
    
    for i, val := range nums {
        j, ok := seen[target - val]
        if ok {
            return []int{j, i}
        }
        seen[val] = i
    }

    return nil
}